app/services/server_config_builder.rb

```diff
@@ -1,41 +1,34 @@
+# frozen_string_literal: true
+
 class ServerConfigBuilder
-  attr_accessor :config
+  attr_accessor :server
 
-  def initialize(server)
+  def initialize(server:)
     @server = server
   end
 
   def generate_config
-    file.each_line do |line|
+    sample_config.each_line do |line|
```

Comment:
We can use ERB for sample config, to substitute variables without manual parsing